### PR TITLE
Pas plumbing II

### PR DESCRIPTION
**Pull request by nickdechant**

### CODE REVIEW

**Comment by jinruh** [comment]
Can you paste some screenshot here?

**Comment by nickdechant** [comment]
screen shot 2016-06-01 at 4 48 14 pm

File: deploy-service/common/src/main/resources/sql/deploy.sql [hunk 1]
```diff
@@ -444,9 +444,18 @@ CREATE TABLE IF NOT EXISTS tags (
...
+ CREATE INDEX tags_target_type_idx ON tags(target_type, created_date)
+
+ CREATE TABLE IF NOT EXISTS pas_configs (
+    group_name VARCHAR(64) NOT NULL,
+    throughput INT,
+    metric VARCHAR(512),
+    last_updated BIGINT NOT NULL,
+    pas_state VARCHAR(32) NOT NULL,
+    PRIMARY KEY (group_name)
+ ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
Comment (yujunglo) [inline deploy-service/common/src/main/resources/sql/deploy.sql]: duplicated table

File: deploy-service/common/src/main/java/com/pinterest/arcee/bean/PasConfigBean.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    @JsonProperty("throughput")
+    private int throughput;
+
+    @NotEmpty
+    @JsonProperty("metric")
+    private String metric;
+
+    @NotEmpty
+    @JsonProperty("pas_state")
+    private String pas_state;
```
Comment (yujunglo) [inline deploy-service/common/src/main/java/com/pinterest/arcee/bean/PasConfigBean.java]: it's better to create a enum state for it

File: deploy-board/deploy_board/templates/groups/asg_config.html [hunk 3]
```diff
@@ -39,6 +40,16 @@
...
     </div>
+
+    <div id="pasConfigPid" class="panel panel-default">
+        {% include "groups/pase_config.tmpl" %}
+    </div>
+
+
+<div id="asgConfigPid" class="panel panel-default">
+{% include "groups/asg_config.tmpl" %}
+</div>
```
Comment (yujunglo) [inline deploy-board/deploy_board/templates/groups/asg_config.html]: duplicated div

File: deploy-board/deploy_board/templates/groups/asg_config.html [hunk 4]
```diff
@@ -39,6 +40,16 @@
     <div id="asgConfigPid" class="panel panel-default">
         {% include "groups/asg_config.tmpl" %}
     </div>
+
+    <div id="pasConfigPid" class="panel panel-default">
+        {% include "groups/pase_config.tmpl" %}
+    </div>
```
Comment (yujunglo) [inline deploy-board/deploy_board/templates/groups/asg_config.html]: I think it's better to put this div in asg_config.tmpl. The alarm or time based auto scaling page is created in asg_config.tmpl based on "autoScalingType". And this pasConfig should be part of alarm based asg.

File: deploy-service/common/src/main/java/com/pinterest/arcee/handler/GroupHandler.java [hunk 5]
```diff
@@ -84,7 +78,9 @@ public GroupHandler(ServiceContext serviceContext) {
         jobPool = serviceContext.getJobPool();
         hostInfoDAO = serviceContext.getHostInfoDAO();
         spotAutoScalingDAO = serviceContext.getSpotAutoScalingDAO();
+        pasConfigDAO = serviceContext.getPasConfigDAO();
         awsVmManager = new AwsVmManager(serviceContext);
+        pasConfigDAO = serviceContext.getPasConfigDAO();
```
Comment (yujunglo) [inline deploy-service/common/src/main/java/com/pinterest/arcee/handler/GroupHandler.java]: duplicated assignment

**Comment by jinruh** [comment]
r+

**Pull request merged**